lib/src/client/transport/xhr_transport.dart

```diff
@@ -180,8 +180,9 @@ class XhrClientConnection extends ClientConnection {
 
   @override
   Future<void> terminate() async {
-    for (XhrTransportStream request in _requests) {
-      request.terminate();
+    while (_requests.isNotEmpty) {
+      final XhrTransportStream request = _requests.first;
```

Comment:
I would suggest:

```dart
while (_requests.isNotEmpty) {
  final requests = List.of(_requests);
  _requests.clear();

    
  await Future.wait(requests.map((rq) => rq.terminate()));
}
```

this would `terminate()` all requests in parallel rather then sequentially.

Reply:
@mraleph AFAIK, there is no need to wait for the requests to terminate. It was just the nature of the loop since I didn't want to copy the requests list. I made the change to copy the list of requests and call terminate on them.

Reply:
@mraleph Please hold on before merging this. I've been running https://github.com/isaldana/grpc-dart for a while and I just realized that I forgot to include https://github.com/isaldana/grpc-dart/commit/0d23114b449f4a5e65d170c6471ab9a9ccc05d3d in this PR.